Description: Initialise DBus threading support
 colord itself uses GDBus, which is threadsafely implemented on top of libdbus.
 .
 However, colord links to other libraries which use libdbus directly, and do so
 from multiple threads.
 .
 Explicitly initialise dbus threading support to avoid weird and wonderful dbus
 crashes inside avahi.
Author: Christopher James Halse Rogers <raof@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/colord/+bug/844286
Forwarded: Not needed (libsane support removed upstream; possibly will be reinstated with this patch)

Index: colord/configure.ac
===================================================================
--- colord.orig/configure.ac	2012-09-12 16:30:07.000000000 +1000
+++ colord/configure.ac	2012-09-12 16:30:07.000000000 +1000
@@ -128,6 +128,8 @@
 dnl ---------------------------------------------------------------------------
 PKG_CHECK_MODULES(GLIB, glib-2.0 >= 2.28.0 gobject-2.0 gthread-2.0 gio-2.0 >= 2.25.9 gio-unix-2.0 gmodule-2.0)
 PKG_CHECK_MODULES(GUSB, gusb >= 0.1.1, have_gusb=yes, have_gusb=no)
+PKG_CHECK_MODULES(DBUS, dbus-1)
+
 if test x$have_gusb = xyes; then
 	AC_DEFINE(HAVE_GUSB,1,[whether gusb is available])
 fi
@@ -184,9 +186,6 @@
 AC_ARG_ENABLE(examples, AS_HELP_STRING([--enable-examples],[enable DBus example code]),
 	      enable_examples=$enableval,enable_examples=yes)
 AM_CONDITIONAL(CD_BUILD_EXAMPLES, test x$enable_examples = xyes)
-if test x$enable_examples = xyes; then
-	PKG_CHECK_MODULES(DBUS, dbus-1)
-fi
 
 dnl ---------------------------------------------------------------------------
 dnl - Build reverse engineering
Index: colord/contrib/colord-sane/Makefile.am
===================================================================
--- colord.orig/contrib/colord-sane/Makefile.am	2012-09-12 16:30:07.000000000 +1000
+++ colord/contrib/colord-sane/Makefile.am	2012-09-12 16:30:07.000000000 +1000
@@ -7,6 +7,7 @@
 	$(GLIB_CFLAGS)					\
 	$(GUDEV_CFLAGS)					\
 	$(SANE_CFLAGS)					\
+	$(DBUS_CFLAGS)					\
 	-I$(top_srcdir)					\
 	-I$(top_srcdir)/libcolord			\
 	-DG_LOG_DOMAIN=\"CdSane\"			\
@@ -32,7 +33,8 @@
 	$(COLORD_LIBS)					\
 	$(GUDEV_LIBS)					\
 	$(SANE_LIBS)					\
-	$(GLIB_LIBS)
+	$(GLIB_LIBS)					\
+	$(DBUS_LIBS)
 
 colord_sane_CFLAGS =					\
 	$(WARNINGFLAGS_C)
Index: colord/contrib/colord-sane/cd-main.c
===================================================================
--- colord.orig/contrib/colord-sane/cd-main.c	2012-09-12 16:30:07.000000000 +1000
+++ colord/contrib/colord-sane/cd-main.c	2012-09-12 16:30:07.000000000 +1000
@@ -26,6 +26,7 @@
 #include <locale.h>
 #include <sane/sane.h>
 #include <gudev/gudev.h>
+#include <dbus/dbus.h>
 
 #include "cd-client.h"
 #include "cd-device.h"
@@ -571,6 +572,7 @@
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 	textdomain (GETTEXT_PACKAGE);
 
+	dbus_threads_init_default ();
 	g_type_init ();
 
 	/* TRANSLATORS: program name */
